explorer/testdata/for/for-loop.carbon

```diff
@@ -0,0 +1,52 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
We should also have some tests that cover `break` and `continue`

Reply:
On the same topic, it would also be nice to have a test with an empty array

Reply:
tests for break and continue are added

Reply:
I agree with @clavin that we should also test iterating over an empty array.